src/openeo_aggregator/backend.py

```diff
@@ -672,14 +677,60 @@ def _get_connection_and_backend_service_id(
-        # TODO: add caching. Also see https://github.com/Open-EO/openeo-aggregator/issues/78#issuecomment-1326180557
+        cached_info = self._memoizer.get_or_call(key=("_get_service_types",), callback=self._get_service_types)
+        # Convert the cached results back to the format that service_types should return.
+        return {name: data["service_type"] for name, data, in cached_info.items()}
+
+    def _find_backend_id_for_service_type(self, service_type: str) -> Optional[str]:
+        """Returns the ID of the backend that provides the service_type or None if no backend supports that service."""
+        cached_info = self._memoizer.get_or_call(key=("_get_service_types",), callback=self._get_service_types)
+        return cached_info.get(service_type, {}).get("backend_id")
+
+    def _get_service_types(self) -> dict:
+        """Returns a dict that maps the service name to another dict that contains 2 items:
+        1) the backend_id that provides this secondary service
+        2) the service_types for self.service_types.
+
+        For example:
+        {'WMTS':
+            {'backend_id': 'b1',
+             'service_type':   // contains what backend b1 returned for service type WMTS.
+                {'configuration':
+                    ...
+                }
+            }
+         'WMS':
+            {'backend_id': 'b2',
+             'service_type':
+                {'configuration':
+                    ...
+                }
+            }
+        }
+
+        Info for selecting the right backend to create a new service
+        ------------------------------------------------------------
+
+        We are assuming that there is only one upstream backend per service type.
+        That means we don't really need to merge duplicate service types, i.e. there is no need to resolve
+        some kind of conflicts between duplicate service types.
+
+        So far we don't store info about the backends' capabilities, but in the future
+        we may need to store that to select the right backend when we create a service.
+
+        See: issues #83 and #84:
+            https://github.com/Open-EO/openeo-aggregator/issues/83
+            https://github.com/Open-EO/openeo-aggregator/issues/84
+        """
+
+        # TODO: Issue #85 data about backend capabilities could be added to the service_types data structure as well.
         service_types = {}
 
         # TODO: Instead of merge: prefix each type with backend-id? #83
-        def merge(formats: dict, to_add: dict):
-            for name, data in to_add.items():
-                if name.lower() not in {k.lower() for k in formats.keys()}:
-                    formats[name] = data
+        def merge(services: dict, backend_id: str, new_service: dict):
+            for name, data in new_service.items():
+                if name.lower() not in {k.lower() for k in services.keys()}:
```

Comment:
I would add an `else`  with warning log that there is a name conflict

Reply:
I have added the else clause, in the for loop that was extracted from the removed inner function `merge`.

I  also added a test case to make sure the logging works, because it seems like this we want to be sure we get the warning when we do have conflicting service types.